fastlane/Fastfile

```diff
@@ -1101,3 +1107,37 @@ def trigger_buildkite_release_build(branch:, beta:)
     pipeline_file: 'release-builds.yml'
   )
 end
+
+def inject_buildkite_analytics_environment(xctestrun_path:)
+  require 'plist'
+
+  xctestrun = Plist.parse_xml(xctestrun_path)
+  xctestrun['TestConfigurations'].each do |configuration|
+    configuration['TestTargets'].each do |target|
+      TEST_ANALYTICS_ENVIRONMENT.each do |key|
+        value = ENV.fetch(key)
+        next if value.nil?
+
+        target['EnvironmentVariables'][key] = value
```

Comment:
Regarding token leaking risk which I agree it's not a major issue, can we try setting these env vars as "variable name to be interpreted" string, instead of the env var value (basically like how it's set in the `.xctestplan` file)?

```suggestion
        target['EnvironmentVariables'][key] = "$(#{key})"
```

Reply:
I have a feeling that might not work.

The whole issue with the environment variables is that `xcodebuild` seems not to pass them when it runs tests in the `test-without-building` mode, using instead the values written in the `xctestrun`.

But, that's only my current understanding. I'll put it to the test and see 😄

Reply:
I tried it in  #7482 and it showed that the `$(key)` syntax doesn't work:

<img width="1177" alt="image" src="https://user-images.githubusercontent.com/1218433/184763386-21505f06-8244-4524-98f0-96480397b3ec.png">

Reply:
🤦‍♂️ Of course, I got confused. Thanks for trying it.